oxo.py

```diff
@@ -0,0 +1,127 @@
+class OxoBoard:
+    def __init__(self,x,y):
+        """ The initialiser. Initialise any fields you need here. """
+        self.grid = {(0,0):0,(0,1):0,(0,2):0,(1,0):0,(2,0):0,(1,1):0,(2,2):0,(1,2):0,(2,1):0}
+        self.x = x
+        self.y = y
+        self.gridCounter = 0
+
+
+    def get_square(self, x, y):
+        """ Return 0, 1 or 2 depending on the contents of the specified square. """
+        return self.grid [x,y]
+
+
+
+    def set_square(self, x, y, mark):
+        """ If the specified square is currently empty (0), fill it with mark and return True.
+        If the square is not empty, leave it as-is and return False. """
+        if self.grid [x,y] == 0:
+            self.grid [x,y] = mark
+            return True
+        else:
+            return False
+
+
+
+
+
+    def is_board_full(self):
+        """ If there are still empty squares on the board, return False.
+            If there are no empty squares, return True. """
+        for x in xrange (0,3):
+            for y in xrange(0,3):
+                if self.grid[x,y] == 0:
+                    return False
+                else:
+                    self.gridCounter += 1
+        if self.gridCounter == 9:
+            return True
+
+    def get_winner(self):
+        """ If a player has three in a row, return 1 or 2 depending on which player.
+            Otherwise, return 0. """
+        if self.grid [1,1] != 0:
+            if self.grid [0,0] == self.grid [1,1]:
+                if self.grid [2,2] == self.grid [1,1]:
+                    return True
```

Comment:
The specification (in the doc-string and also in the worksheet) specifies that `get_winner` should return `0`, `1` or `2`, not `True` or `False`. So for example here you should return `self.grid[1,1]` instead of `True`.